memcache/memcache.go

```diff
@@ -667,3 +691,61 @@ func (c *Client) incrDecr(verb, key string, delta uint64) (uint64, error) {
 	})
 	return val, err
 }
+
+func parseStatsLine(s []byte, expectedPrefix []byte) (uint64, error) {
+          if !bytes.HasPrefix(s, expectedPrefix) {
+                    return 0, errors.New("line does not start with exceptedPrefix")
+          }
+          return strconv.ParseUint(
+              string(s[len(expectedPrefix):len(s)-2]), 10, 64)
+}
+
+func parseStatsResponse(r *bufio.Reader, stats *Statistics) error {
+	for {
+		line, err := r.ReadSlice('\n')
+		if err != nil {
+			return err
+		}
+		if bytes.Equal(line, resultEnd) {
+			return nil
+		}
+		if hits, err := parseStatsLine(line, statsGetHits); err == nil {
+			stats.Hits = hits
+			continue
+		}
+		if misses, err := parseStatsLine(line, statsGetMisses); err == nil {
+			stats.Misses = misses
+			continue
+		}
+		if bytesWritten, err := parseStatsLine(line, statsBytesWritten); err == nil {
+			stats.ByteHits = bytesWritten
+			continue
+		}
+		if items, err := parseStatsLine(line, statsItems); err == nil {
+			stats.Items = items
+			continue
+		}
+		if bytes_total, err := parseStatsLine(line, statsBytes); err == nil {
+			stats.Bytes = bytes_total
+			continue
+		}
+	}
+}
+
+
+func (c *Client) Stats(addr net.Addr) (*Statistics, error) {
```

Comment:
This lacks comments.

Also, why does it take an Addr? It seems like most people would want aggregated statistics across all nodes for a method named "Stats".

Maybe this should be called "NodeStats".